alibi_detect/od/_svm.py

```diff
@@ -0,0 +1,236 @@
+        `n` is the number of reference instances and `m` defines the number of inducing points. This can therefore be
+        expensive for large reference sets and benefits from implementation on the GPU.
+
+        In general if using a small dataset then using the `''cpu''` with the optimization `''sgd''` is the best choice.
+        Whereas if using a large dataset then using the `''gpu''` with the optimization `''bgd''` is the best choice.
+
+        Parameters
+        ----------
+        nu
+            The proportion of the training data that should be considered outliers. Note that this does not necessarily
+            correspond to the false positive rate on test data, which is still defined when calling the
+            `infer_threshold` method. `nu` should be thought of as a regularization parameter that affects how smooth
+            the svm decision boundary is.
+        n_components
+            Number of components in the Nystroem approximation, By default uses all of them.
+        kernel
+            Kernel function to use for outlier detection. Should be an instance of a subclass of `torch.nn.Module`. If
+            not specified then defaults to the `GaussianRBF`.
+        optimization
+            Optimization method to use. Choose from ``'sgd'`` or ``'bgd'``. Defaults to ``'sgd'``.
+        backend
+            Backend used for outlier detection. Defaults to ``'pytorch'``. Options are ``'pytorch'``.
+        device
+            Device type used. The default tries to use the GPU and falls back on CPU if needed. Can be specified by
+            passing either ``'cuda'``, ``'gpu'``, ``'cpu'`` or an instance of ``torch.device``.
+
+        Raises
+        ------
+        NotImplementedError
+            If choice of `backend` is not implemented.
+        ValueError
+            If choice of `optimization` is not valid.
+        ValueError
+            If `n_components` is not a positive integer.
+        """
+        super().__init__()
+
+        if optimization not in ('sgd', 'bgd'):
+            raise ValueError(f'Optimization {optimization} not recognized. Choose from `sgd` or `bgd`.')
+
+        if n_components is not None and n_components <= 0:
+            raise ValueError(f'n_components must be a positive integer, got {n_components}.')
+
+        backend_str: str = backend.lower()
+        BackendValidator(
+            backend_options={'pytorch': ['pytorch']},
+            construct_name=self.__class__.__name__
+        ).verify_backend(backend_str)
+
+        backend_cls = backends[backend][optimization]
+        args: Dict[str, Any] = {
+            'n_components': n_components,
+            'kernel': kernel,
+            'nu': nu
+        }
+        args['device'] = device
+        self.backend = backend_cls(**args)
+
+    def fit(
+        self,
+        x_ref: np.ndarray,
+        tol: float = 1e-6,
+        max_iter: int = 1000,
+        step_size_range: Tuple[float, float] = (1e-6, 1.0),
```

Comment:
Can we change default `step_size_range` to `(1e-8, 1.0)`?

Reply:
Sorry, missed this! Now fixed!